packages/common/src/Widget.ts

```diff
@@ -657,7 +661,21 @@ export abstract class Widget extends PropertyExt {
         return this;
     }
 
-    enter(_domNode: HTMLElement, _element: d3SelectionType) { }
+    enter(_domNode: HTMLElement, _element: d3SelectionType) {
+        if ((window as any).__hpcc_debug) {
```

Comment:
I think this code would need its own "switch" as its going to be very very noisy. 
Also it should be done on the constructor and probably in the `Class` class?

Reply:
And maybe only for specific methods?

Reply:
Hmm... maybe another global variable could be used to control which methods are logged?
For example:
```javascript
window.__hpcc_debug_methods = ["enter","update","exit","render","resize","target","columns","data"];
```
...and how they're logged:
```javascript
window.__hpcc_debug_callbacks = {
    "args": methodName=>{
        this._logger.args(methodName, arguments);
    },
    "timestamps": methodName=>{
        this._logger.timestamps(methodName, Date.getTime());
    },
    "trace": methodName=>{
        console.groupCollapsed(this._logger._id+" "+methodName);
        console.trace();
        console.groupEnd();
    }
};
```